core/src/smartcontracts/isi/world.rs

```diff
@@ -352,11 +357,44 @@ pub mod isi {
 
             let raw_validator = self.object;
             let engine = wsv.engine.clone(); // Cloning engine is cheap
+
+            let (new_validator /*, new_permission_token_definitions*/) =
+                || -> Result<_, crate::smartcontracts::wasm::error::Error> {
+                    {
+                        let new_validator = Validator::new(raw_validator, &engine)?;
+                        // let new_permission_token_definitions =
+                        //     new_validator.permission_tokens(wsv)?;
+                        Ok((new_validator /*, new_permission_token_definitions*/))
+                    }
+                }()
+                .map_err(|error| {
+                    InvalidParameterError::Wasm(format!("{:?}", eyre::Report::from(error)))
+                })?;
+
             let world = wsv.world_mut();
-            let new_validator = Validator::new(raw_validator, &engine)
-                .map_err(|err| InvalidParameterError::Wasm(err.to_string()))?;
             let _ = world.upgraded_validator.insert(new_validator);
 
+            // let old_permission_token_definitions = wsv
+            //     .permission_token_definitions()
+            //     .values()
+            //     .cloned()
+            //     .collect::<HashSet<_>>();
+            // let new_permission_token_definitions =
+            //     HashSet::from_iter(new_permission_token_definitions);
+
+            // old_permission_token_definitions
+            //     .difference(&new_permission_token_definitions)
+            //     .map(|definition| Unregister::<PermissionTokenDefinition> {
+            //         object_id: definition.id.clone(),
+            //     })
+            //     .try_for_each(|unregister| unregister.execute(authority, wsv))?;
+
+            // new_permission_token_definitions
+            //     .difference(&old_permission_token_definitions)
+            //     .cloned()
+            //     .map(|definition| Register::<PermissionTokenDefinition> { object: definition })
+            //     .try_for_each(|new| new.execute(authority, wsv))?;
```

Comment:
there is some commented code in this module

Reply:
Yes, that's what I talked about before in the chat. Take a look at `TODO` above in this file.  
This is done on purpose to demonstrate how other code should be used, but I can't uncomment it because analysis is not supported on the wasm side yet